### PR TITLE
Add feature flags to CAPI; update WearOS build steps

### PR DESCRIPTION
**Pull request by Manishearth**
This allows the user to pref on/off various modules of the CAPI crate. By default, nothing changes.

This fixes https://github.com/unicode-org/icu4x/issues/891 by adding a "freertos" feature.

This also adds a "wearos" feature flag to the CAPI crate that contains the current WearOS feature set, for ease of use, including a leaner testdata file that contains just the fixed decimal symbols for bn and en.

Note that this does not propagate to the generated headers; Diplomat does not understand feature flags. Users of the C/CPP/WASM api are expected to avoid any APIs they have preffed off.

Finally, this updates CI to use the appropriate invocation for wearos.

### CODE REVIEW

File: provider/testdata/src/blob.rs [hunk 1]
```diff
@@ -4,11 +4,17 @@
...
     env!("CARGO_MANIFEST_DIR"),
-    "/data/testdata.postcard"
+    "/data/decimal-bn-en.postcard"
 ));
 
+#[cfg(not(feature = "smaller_static"))]
+const STATIC_STR_DATA: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/data/testdata.postcard"
+));
```
Comment (sffc) [inline provider/testdata/src/blob.rs:17]: Issue: I'm concerned that doing it this way could cause problems where any dependency that enables the `"smaller_static"` feature would replace the full data with the smaller data.

Suggestion: Add two functions or modules, one that returns the full data and one that returns the small data.

I don't think they need to be two separate features in `icu_testdata`.  I'd like if the feature could be in `capi` only, and that's where the choice between the two bundles is chosen.

File: ffi/capi/Cargo.toml [hunk 2]
```diff
@@ -26,33 +26,47 @@ include = [
...
 crate-type = ["staticlib", "rlib", "cdylib"]
 path = "src/lib.rs"
 
+[features]
+default = ["plurals", "decimal", "locale_canonicalizer", "provider_fs", "provider_static"]
+wearos = ["decimal", "smaller_static", "freertos"]
+
+plurals = ["icu_plurals"]
+decimal = ["icu_decimal", "fixed_decimal"]
+locale_canonicalizer = ["icu_locale_canonicalizer"]
```
Comment (sffc) [inline ffi/capi/Cargo.toml]: Request: can we avoid adding feature flags for components?  That leads to an unbounded list of features in the future, which comes with a nontrivial maintenance cost.

**Review by sffc — CHANGES_REQUESTED** [review]

File: provider/testdata/src/blob.rs (hunk 1 above)
Comment (Manishearth) [reply provider/testdata/src/blob.rs:17]: sounds good

File: provider/testdata/Cargo.toml [hunk 3]
```diff
@@ -156,4 +156,5 @@ metadata = [
 
 fs = ["icu_provider_fs", "std"]
 static = ["icu_provider_blob"]
+# Uses a smaller set of testdata for the static data provider (only decimal data for bn/en)
```
Comment (sffc) [inline provider/testdata/Cargo.toml:159]: Nit: Obsolete line

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**